feat: Add multi-select support for media assets

viva-jinyi opened the pull request:
## Summary
Implements file explorer-style multi-selection functionality for media assets in the AssetsSidebarTab component.

## Changes

### Multi-Selection Interactions
- **Normal click**: Single selection (clears previous, selects new)
- **Shift + click**: Range selection (from last selected to current)
- **Ctrl/Cmd + click**: Toggle individual selection

### State Management
- Added `assetSelectionStore` to manage selected asset IDs using Set
- Created `useAssetSelection` composable for selection logic and keyboard state

### UI Enhancements
- Display selection count in footer (output tab only)
- Interactive selection count that shows "Deselect all" on hover
- Added bulk action buttons for download/delete (UI only)

### Translation Keys
Added new keys under `mediaAsset.selection`:
- `selectedCount`: "{count} selected"
- `deselectAll`: "Deselect all"
- `downloadSelected`: "Download"
- `deleteSelected`: "Delete"

## Test Plan
- [x] Open Assets sidebar tab
- [x] Switch to Generated tab
- [x] Test single selection with normal click
- [x] Test range selection with Shift + click
- [x] Test toggle selection with Ctrl/Cmd + click
- [x] Verify selection count updates correctly
- [x] Test hover interaction on selection count
- [x] Click "Deselect all" to clear selection
- [x] Test bulk action buttons (UI only)

## Notes
- Bulk download/delete functionality to be implemented in separate PR
- Selection UI currently only shows in output (Generated) tab

[screen-capture.webm](https://github.com/user-attachments/assets/740315bd-9254-4af3-a0be-10846d810d65)

viva-jinyi committed:
feat: Implement multi-select functionality for media assets

- Add multi-select interactions with Shift/Ctrl(Cmd) keys
- Normal click: Single selection
- Shift + click: Range selection
- Ctrl/Cmd + click: Individual multi-selection
- Create assetSelectionStore for managing selected assets
- Encapsulate selection logic in useAssetSelection composable
- Add selection count display and bulk action buttons in footer
- Shows "Deselect all" on hover
- Download/Delete buttons (actual implementation pending)

viva-jinyi committed:
chore: moved to platform/assets

viva-jinyi committed:
fix: Address code review feedback for multi-select

- Performance: Use batch operations for range selection instead of forEach
- Memory: Add reset function to clear selection on component unmount
- Security: Add input validation for asset operations
- Accessibility: Add keyboard support for selection count interaction
- Optimization: Prevent unnecessary re-renders in Set operations

viva-jinyi committed:
fix: Reset hover state after deselect all action

- Clear isHoveringSelectionCount when deselecting all assets
- Prevents 'Deselect all' text from staying visible after clearing selection
- Improves UX by returning to normal selection count display

viva-jinyi commented on src/platform/assets/composables/useAssetSelectionStore.ts:
```diff
@@ -0,0 +1,67 @@
+
+  return {
+    // State
+    selectedAssetIds: computed(() => selectedAssetIds.value),
```
✅

viva-jinyi commented on src/platform/assets/composables/useAssetSelectionStore.ts:
```diff
@@ -0,0 +1,67 @@
+
+export const useAssetSelectionStore = defineStore('assetSelection', () => {
+  // State
+  const selectedAssetIds = ref<Set<string>>(new Set())
```
✅

christian-byrne commented on src/platform/assets/composables/useAssetSelectionStore.ts:
nit (non-blocking): It might make more sense to scope this to the lifecycle of the asset sidebar

viva-jinyi commented on src/components/sidebar/tabs/AssetsSidebarTab.vue:
```diff
@@ -200,18 +274,15 @@ const refreshAssets = async () => {
-  } else {
-    selectedAsset.value = asset
-  }
+  const index = displayAssets.value.findIndex((a) => a.id === asset.id)
```
✅

viva-jinyi committed:
feat: Add multi-select functionality and improve folder view for media assets

  - Add bulk download/delete for multiple selected assets
  - Hide delete buttons in folder view
  - Activate keyboard shortcuts only when sidebar is open
  - Add activate/deactivate lifecycle methods to useAssetSelection

viva-jinyi commented:
### 🎯 Multi-Select Improvements
- **Bulk Download**: Download multiple selected assets at once
- **Bulk Delete**: Delete selected assets with confirmation dialog
- Added `downloadMultipleAssets` and `deleteMultipleAssets` methods to `useMediaAssetActions`

### 🔒 Folder View Restrictions
- Individual delete buttons are now hidden in folder view
- Bulk delete button is hidden in folder view
- Controlled via `show-delete-button` prop from parent component

### ⌨️ Keyboard Event Optimization
- Shift/Ctrl key events only active when asset sidebar is open
- Lifecycle managed via `activate()`/`deactivate()` methods
- Selection state automatically resets when sidebar closes
